config/config_mgmt.py

```diff
@@ -253,7 +253,77 @@ def __init__(self, source="configDB", debug=False, allowTablesWithoutYang=True):
     def __del__(self):
         pass
 
-    def _checkKeyinAsicDB(self, key, db):
+    def _checkPortsDownAppDb(self, db, ports):
+        '''
+        Check APP DB for PORT oper status down.
+
+        Parameters:
+            db (SonicV2Connector): database.
+            ports (list): List of ports
+
+        Returns:
+            (bool): True, if all ports are oper down.
+        '''
+        try:
+            # connect to APP DB,
+            db.connect(db.APPL_DB)
+            for port in ports:
+                # Key format "PORT_TABLE:Ethernet112"
+                key = 'PORT_TABLE:{}'.format(port)
+                portOperStatus = db.get('APPL_DB', key, 'oper_status')
+                self.sysLog(syslog.LOG_DEBUG, "APPL_DB: {} {}".\
+                    format(key, portOperStatus))
+                if portOperStatus is None or portOperStatus != 'down':
+                    return False
+
+        except Exception as e:
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
+            return False
+
+        return True
+
+    def _verifyAppDB(self, db, ports, timeout):
+        '''
+        Verify in the App DB that ports are oper_status down,
+        Keep on trying till timeout period.
+
+        Parameters:
+            db (SonicV2Connector): database.
+            ports (list): List of ports
+            timeout (int): timeout period.
+
+        Returns:
+            (bool): True, if all ports are oper down.
+        '''
+        self.sysLog(msg="Verify Port oper_status down from App DB, Wait...")
+
+        try:
+            for waitTime in range(timeout):
+                self.sysLog(logLevel=syslog.LOG_DEBUG, \
+                    msg='Check App DB: {} try'.format(waitTime+1))
+                # checkPortsDownAppDb will return True if all ports are Oper
+                # down in APP DB
+                if self._checkPortsDownAppDb(db, ports):
+                    break
+                tsleep(1)
+
+            # raise if timer expired
+            if waitTime + 1 == timeout:
+                self.sysLog(logLevel=syslog.LOG_CRIT, \
+                    msg="!!! Critical Failure, Ports are not yet Operation \
+                    Down in App DB, Bail Out !!!",  doPrint=True)
+                raise(Exception("Ports are not Operation Down in App DB after {} secs"\
+                    .format(timeout)))
+
+        except Exception as e:
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, \
```

Comment:
>sysLog [](http://example.com/codeflow?start=17&length=6)

`syslog`

Reply:
Here, self.sysLog is a class method that calls syslog lib funcs.

Reply:
I mean the captalization `syslog` is more natural.

Reply:
Agree with @qiluo-msft. In SONiC, our Python code style follows the [PEP8 style guidelines](https://www.python.org/dev/peps/pep-0008/#naming-conventions). However, the config_mgmt.py and config_mgmt_test.py stray from this convention. I suggest you plan for a future PR to align the style with the rest of the SONiC codebase.

Reply:
This comment is still applicable. `sysLog` looks weird, suggest `syslog`.